# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/ipc/RemoteImageHolder.h
# Commit: 2a830d3c1fb0
# Full Hash: 2a830d3c1fb01d887bd9ec41aeb8b51dbc0bfc62
# Author: Brad Werth <bwerth@mozilla.com>
# Date: 2022-09-09 04:30:58
# Regressor Bug: 1787561
# File Overlap Count: 1
# Description:
#   Bug 1787561 Part 1: Make GPUVideoImage track color depth, and make RemoteImageHolder pass it through. r=media-playback-reviewers,gfx-reviewers,alwu,aosmond
#   
#   This ensures that the Image that the client receives has the color depth
#   information that was sent to the GPU, even though the client can't query
#   the GPU directly.
# ==============================================================================

diff -r edaed3aeb0f9 -r 2a830d3c1fb0 dom/media/ipc/RemoteImageHolder.h
--- a/dom/media/ipc/RemoteImageHolder.h	Thu Sep 08 17:55:07 2022 +0000
+++ b/dom/media/ipc/RemoteImageHolder.h	Thu Sep 08 18:02:03 2022 +0000
@@ -28,6 +28,7 @@
   RemoteImageHolder(layers::IGPUVideoSurfaceManager* aManager,
                     layers::VideoBridgeSource aSource,
                     const gfx::IntSize& aSize,
+                    const gfx::ColorDepth& aColorDepth,
                     const layers::SurfaceDescriptor& aSD);
   RemoteImageHolder(RemoteImageHolder&& aOther);
   // Ensure we never copy this object.
@@ -47,6 +48,7 @@
   // We need a default for the default constructor, never used in practice.
   layers::VideoBridgeSource mSource = layers::VideoBridgeSource::GpuProcess;
   gfx::IntSize mSize;
+  gfx::ColorDepth mColorDepth;
   Maybe<layers::SurfaceDescriptor> mSD;
   RefPtr<layers::IGPUVideoSurfaceManager> mManager;
 };